### PR TITLE
Updated release pipeline script to create tags for releases off patch branches in the format: 20190730.1-patch1

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: vsts/pipelines/ci.yml [hunk 1]
```diff
@@ -5,6 +5,30 @@ variables:
 - group: Oryx
 
 stages:
+  - stage: CreateReleaseTag
```
Comment (kichalla) [inline vsts/pipelines/ci.yml]: Idea is to create the release tag name and use it across all stages in the pipeline but it seems ADO does not have a in-built way to pass in data between stages , so here I am writing out the release tag name to an artifact file which will be later read and set as environment variable in other stages.

File: vsts/scripts/createReleaseTag.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    # Name of the tag which is being patched (ex: 20190730.1)
+    patchedTagName=${branchNameParts[1]}
+
+    # We want tags for patch releases in the format: 20190730.1-patch1, 20190730.1-patch2 etc.
+    baseReleaseTagUrl="https://github.com/microsoft/Oryx/releases/tag"
+    patchNumber=0
+
+    # Increment patch numbers until we find one for which we have not created a release already
+    while true; do
+        patchNumber=$((patchNumber + 1))
```
Comment (kichalla) [inline vsts/scripts/createReleaseTag.sh]: Sample releases created with these changes:
image

File: build/buildBuildImages.sh [hunk 3]
```diff
@@ -92,7 +94,7 @@ function buildDockerImage() {
 	# Retag build image with build number tags
 	if [ "$AGENT_BUILD" == "true" ]
 	then
-		uniqueTag="$BUILD_DEFINITIONNAME.$BUILD_NUMBER"
+		uniqueTag="$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME"
```
[the remark this answers is not in the document]
Comment (gpcastro) [reply build/buildBuildImages.sh]: should we add these PR comments as actual comments in the code? They really help understand what is going on

[the remark this answers is not in the document]
Comment (kichalla) [reply build/buildBuildImages.sh]: Sure, will do

**Pull request merged**